openmdao/drivers/differential_evolution_driver.py

```diff
@@ -0,0 +1,483 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import copy
+import numpy as np
+import openmdao
+
+from openmdao.core.analysis_error import AnalysisError
+from openmdao.core.driver import Driver, RecordingDebugging
+from openmdao.utils.mpi import MPI
+from six import iteritems, itervalues, next
+
+try:
+    from tqdm import tqdm
+except ModuleNotFoundError:
+
+    def tqdm(i, _):
+        return i
+
+
+from differential_evolution import DifferentialEvolution, EvolutionStrategy
+
+if not MPI:
+    rank = 0
+else:
+    rank = MPI.COMM_WORLD.rank
+
+
+class DifferentialEvolutionDriver(Driver):
+    def __init__(self, **kwargs):
```

Comment:
BTW, there are a number of lint failures reported by our CI. Some of them are due to the missing class docstring, which should contain Attributes section for defining class variables (see SimpleGADriver for example).  If you run the tests in openmdao/code_review, you can see the problems locally without having to retrigger the CI. Of course, we always recommend running all the tests.

Reply:
Generally, it's clear I have some tidying up to do. I will get to work on this.

Reply:
Added docstrings in 43fc6af0a4ed38c8e734bf4e0025673f362219fb